sqlmesh/utils/pydantic.py

```diff
@@ -10,7 +10,7 @@ class PydanticModel(BaseModel):
     class Config:
         arbitrary_types_allowed = True
         extra = "forbid"
-        json_encoders = {exp.Expression: lambda e: e.sql(dialect=e.meta.get("dialect"))}
+        json_encoders = {exp.Expression: lambda e: e.meta.get("sql") or e.sql()}
```

Comment:
don't you need to do sql(dialect?

Reply:
Assuming the `e.sql()` part is reached only for python / seed models (see https://github.com/TobikoData/sqlmesh/pull/1187#discussion_r1269807721), then this could be consistent with what we did before because those models don't have raw SQL associated with them and so `meta` would not have a `"dialect"` key set.

Now, I'm not sure, but there might be a difference w.r.t. the model meta properties.

Reply:
yea, this is probably what caused your previous bug with the partitions

Reply:
Do we want to keep `dialect` as a fallback strategy? Eg. if we modify the original expression we might want to reset the `sql` attribute but keep the `dialect` one

Reply:
sorry i'm unclear what you mean

Reply:
> Do we want to keep dialect as a fallback strategy? Eg. if we modify the original expression we might want to reset the sql attribute but keep the dialect one

@izeigerman do you mean for the SQL models too, as in keep the `meta["dialect"] = ...` logic in `dialect.py`? If that's what you're suggesting, I understand that we'd instead do:

```python
lambda self, e: e.meta["sql"] or e.sql(dialect=e.meta.get("dialect"))
```

Reply:
I went ahead and implemented this: seemed like a small overhead to preserve `"dialect"` in the `meta` dict to have a safe fallback which is basically what we did before. What do you think?